c2rust-analyze/src/dataflow/type_check.rs

```diff
@@ -109,7 +109,14 @@ impl<'tcx> TypeChecker<'tcx, '_> {
 
         match *rv {
             Rvalue::Use(ref op) => self.visit_operand(op),
-            Rvalue::Repeat(..) => todo!("visit_rvalue Repeat"),
+            Rvalue::Repeat(ref op, _) => {
+                assert!(matches!(lty.kind(), TyKind::Array(..)));
+                assert_eq!(lty.args.len(), 1);
+                let elem_lty = lty.args[0];
```

Comment:
nit: I've noticed this in general.  I think we can do something like an `assert_matches!` here and destructure the array instead.

Reply:
831aad17